source/common/upstream/cluster_factory_impl.cc

```diff
@@ -31,6 +31,11 @@ ClusterFactoryImplBase::create(const envoy::config::cluster::v3::Cluster& cluste
        ProtobufWkt::Struct::GetDescriptor()->full_name())) {
     cluster_config_type_name =
         TypeUtil::typeUrlToDescriptorFullName(cluster.cluster_type().typed_config().type_url());
+    // This should be behind a runtime flag.
+    if (cluster_config_type_name == "envoy.cluster.strict_dns" ||
```

Comment:
This may not be needed, because there should be no configs out there using these legacy names in a custom cluster type.